src/main/java/com/epam/aidial/core/security/AccessService.java

```diff
@@ -52,6 +53,28 @@ public boolean hasWriteAccess(ResourceDescription resource, ProxyContext context
         return isMyResource(resource, context) || isAppResource(resource, context) || hasReviewAccess(resource, context, false);
     }
 
+    public boolean hasPublicAccess(List<ResourceDescription> resources, ProxyContext context) {
+        boolean isAllPublic = resources.stream().allMatch(ResourceDescription::isPublic);
+        if (!isAllPublic) {
+            throw new IllegalArgumentException("Provided resources must be public");
+        }
+
+        return ruleService.hasPublicAccess(context, resources);
+    }
+
+    private boolean hasPublicAccess(ResourceDescription resource, ProxyContext context, boolean readOnly) {
```

Comment:
Could you please add a javadoc?
These two methods have the same name but do the job differently.
Why can we not marge the methods something like that?
```
  hasPublicAccess(List<Resource> resources, ProxyContext context) {
    for (var resource : resources) {
         if (!hasPublicAcess(resource, context) {
             return false;
         }
   }
   return true;
}
```

Reply:
refactored